bin/prepare-release.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Prepare a new release by updating the version numbers in all related files,
+# updating the changelog to include the released version.
+#
+# Optionally, it can create the release commit and tag it.
+#
+# Usage: prepare-release.sh <new version>
+#
+# Environment:
+#
+# DRY_RUN - Do not modify files or run destructive commands when set.
+# VERB    - Log commands that are run when set.
+
+set -eo pipefail
+
+run() {
+  [ "${VERB:-0}" != 0 ] && echo "+ $*"
+  if [ "$DOIT" = y ]; then
+      "$@"
+  else
+      true
+  fi
+}
+
+# All sed commands below work with either
+# GNU sed (standard on Linux distrubtions) or BSD sed (standard on macOS)
+SED="sed"
+
+WORKSPACE_ROOT="$( cd "$(dirname "$0")/.." ; pwd -P )"
+
+if [ -z "$1" ]; then
+    echo "Usage: $(basename "$0") <new version>"
+    echo
+    echo "Prepare for a new release by setting the version number"
+    exit 1
+fi
+
+NEW_VERSION="$1"
+DATE=$(date +%Y-%m-%d)
+
+if ! echo "$NEW_VERSION" | grep --quiet --extended-regexp '^[0-9]+\.[0-9]+\.[0-9]+(-[a-z0-9.-]+)?'; then
+    echo "error: Specified version '${NEW_VERSION}' doesn't match the Semantic Versioning pattern."
+    echo "error: Use MAJOR.MINOR.PATCH versioning."
+    echo "error: See https://semver.org/"
+    exit 1
+fi
+
+echo "Preparing update to v${NEW_VERSION} (${DATE})"
+echo "Workspace root: ${WORKSPACE_ROOT}"
+echo
+
+GIT_STATUS_OUTPUT=$(git status --untracked-files=no --porcelain)
+if [ -z "$ALLOW_DIRTY" ] && [ -n "${GIT_STATUS_OUTPUT}" ]; then
+    lines=$(echo "$GIT_STATUS_OUTPUT" | wc -l | tr -d '[:space:]')
+    echo "error: ${lines} files in the working directory contain changes that were not yet committed into git:"
+    echo
+    echo "${GIT_STATUS_OUTPUT}"
+    echo
+    echo 'To proceed despite this and include the uncommited changes, set the `ALLOW_DIRTY` environment variable.'
+    exit 1
+
+fi
+
+DOIT=y
+if [[ -n "$DRY_RUN" ]]; then
+    echo "Dry-run. Not modifying files."
+    DOIT=n
+fi
+
+### Rollup files ###
+
+# Update the version in the package.json file
+
+FILE=package.json
+run $SED -i.bak -E \
+    -e "s/^\"version\": \"[0-9a-z.-]+\"/\"version\": \"${NEW_VERSION}\"/" \
+    "${WORKSPACE_ROOT}/${FILE}"
```

Comment:
I think this is OK for now, have you thought about writing in JS so we can use nicer tools (like a real JSON parser? :))

Reply:
Heh, I'm going for the least possible amount of work here :) I'm re-using what the Glean team did and just slightly changing it to make it work for us. We could definitely write/do something different/better when we'll have more time!